### ISSUE TITLE
Menu categories in english without icons

### ISSUE THREAD

**Issue by stefonarch** [lxqt/lxqt#1679]
First installed LXQt from arch repos, then removed and compiled from git, not sure when this happened. All category names  are not translated.
##### Expected Behavior

Menu categories should be localized.
##### Current Behavior

They aren't here.

##### System Information

Up-to-date  arch and LXQt git checkouts

![schermata-17-18-35-16](https://user-images.githubusercontent.com/10681413/52916861-de003a80-32e4-11e9-917d-ec1049b06359.png)

**Comment by agaida** [comment]
pacman -S qt5translations ??

**Comment by stefonarch** [comment]
Is installed

**Comment by agaida** [comment]
one should install and set a matching icon theme - papirus is fine, oxygen too - check 
if lxqt-qtplugin is installed - btw, maybe a build order issue?

**Comment by stefonarch** [comment]
Build order was always the same
` libqtxdg \
        lxqt-build-tools \
        liblxqt \
        libsysstat \
        compton-conf \
        libfm-qt \
        lxqt-themes \
        lxqt-qtplugin \
        pavucontrol-qt \
        qtermwidget \
        qterminal \
        lxqt-about \
        lxqt-admin \
        lxqt-config \
        lxqt-policykit \
        lxqt-powermanagement \
        lxqt-globalkeys \
        lxqt-openssh-askpass \
        lxqt-notificationd \
        lxqt-appswitcher \
        lxqt-sudo \
        lxqt-session \
        pcmanfm-qt \
        lxqt-panel \
        lxqt-runner \
        lxqt-archiver \
        compton-conf \
        qps \
        obconf-qt

`

Breeze, papirus, nothing works and the english issue isn't related to theme...

**Comment by tsujan** [comment]
I just hope this is an Arch build problem and related to https://github.com/lxqt/lxqt/issues/1676 -- but I'm not saying it is so.

**Comment by stefonarch** [comment]
Solved. I tried  to reinstall lxqt-panel from arch repos and it wanted to install also `lxmenu-data`, so this one should be added as dependency in lxqt-panel.

**Comment by tsujan** [comment]
I don't think `lxmenu-data` was the cause but have no intention to search for one.

**Comment by agaida** [comment]
it is a runtime dependency - and we should re-open and retitle or open a new bug about.
my guess is that we use these datas:
```
lxmenu-data: /etc/xdg/menus/lxde-applications.menu
lxmenu-data: /usr/share/desktop-directories/lxde-audio-video.directory
lxmenu-data: /usr/share/desktop-directories/lxde-debian.directory
lxmenu-data: /usr/share/desktop-directories/lxde-development.directory
lxmenu-data: /usr/share/desktop-directories/lxde-education.directory
lxmenu-data: /usr/share/desktop-directories/lxde-game.directory
lxmenu-data: /usr/share/desktop-directories/lxde-graphics.directory
lxmenu-data: /usr/share/desktop-directories/lxde-hardware.directory
lxmenu-data: /usr/share/desktop-directories/lxde-internet-and-network.directory
lxmenu-data: /usr/share/desktop-directories/lxde-look-and-feel.directory
lxmenu-data: /usr/share/desktop-directories/lxde-menu-applications.directory
lxmenu-data: /usr/share/desktop-directories/lxde-menu-system.directory
lxmenu-data: /usr/share/desktop-directories/lxde-network.directory
lxmenu-data: /usr/share/desktop-directories/lxde-office.directory
lxmenu-data: /usr/share/desktop-directories/lxde-other.directory
lxmenu-data: /usr/share/desktop-directories/lxde-personal.directory
lxmenu-data: /usr/share/desktop-directories/lxde-science-math.directory
lxmenu-data: /usr/share/desktop-directories/lxde-settings-system.directory
lxmenu-data: /usr/share/desktop-directories/lxde-settings.directory
lxmenu-data: /usr/share/desktop-directories/lxde-system-tools.directory
lxmenu-data: /usr/share/desktop-directories/lxde-system.directory
lxmenu-data: /usr/share/desktop-directories/lxde-utility-accessibility.directory
lxmenu-data: /usr/share/desktop-directories/lxde-utility.directory
``` 
so it would be cool if we could remove this ancient dependency

**Comment by stefonarch** [comment]
> I don't think `lxmenu-data` was the cause but have no intention to search for one.

I can tell you: `pacman -S menu-data` and I opened the menu and saw icons and italian text appearing live.

**Comment by stefonarch** [comment]
> ```
....> it is a runtime dependency - and we should re-open and retitle or open a new bug about.
> ....
> so it would be cool if we could remove this ancient dependency

Put those files in liblxqt or where it fits well?

**Comment by tsujan** [comment]
If so, this should be reopened until the problem is fixed. I'll change the title.

**Comment by yan12125** [comment]
From https://github.com/lxqt/lxqt-panel/pull/1868#issuecomment-1448574948:

> some parts came out messy and I stopped.

Do you remember what were the issues?

**Comment by stefonarch** [comment]
> Do you remember what were the issues?

No, I don't remember, it was mostly a search-and-replace lxde><lxqt . It _shouldn't_ be difficult.

**Comment by stefonarch** [comment]
Ok, looked into it, looks like the only issue was that the menus should also replace lxde → lxqt so it needs an updated lxqt-panel.

To test: 
1. `sudo rm /usr/share/desktop-directories/lxde-*`
2. Install https://github.com/stefonarch/lxqt-menu-data
3. Open `/etc/xdg/menus/lxqt-applications.menu 
4. search and replace "lxde" with "lxqt" and save.

**Comment by yan12125** [comment]
Thanks! Sounds like an import is feasible.

> If we adopt it it could be directly shipped with the panel?

If one day the issue around menu-cache (https://github.com/lxqt/libfm-qt/issues/417) is resolved, libfm-qt may need those files as well. Keeping them as a separate project (like what you have done) is better.

**Comment by stefonarch** [comment]
PR is here: https://github.com/lxqt/lxqt-panel/pull/1869
`pcmanfm-qt` needs it afaik too.

**Comment by tsujan** [comment]
> pcmanfm-qt needs it afaik too.

pcmanfm-qt is condemned to use `lxmenu-data`, because of `menu-cache`.

I'm busy now, so I may be wrong. But if lxqt-panel can still be compiled with `menu-cache`, the replacement won't be so easy.

**Comment by stefonarch** [comment]
I'm not sure if the names of the menus (applications.menu|lxde-application.menu|lxqt-application.menu) make any difference to menu-cache. Looks like starting once `menu-cached` after the change fixed folder names and icons in `menu://applications`

![screen_area_gio_18:39:28_](https://user-images.githubusercontent.com/10681413/222509498-52575e03-6204-4c6d-aa6e-ebc8c4ecca77.png)

But here I see a [prefix to lxde](https://github.com/lxqt/libfm-qt/blob/d79919165fbda4b6a9cd1c49e3eceee69a127900/src/appmenuview.cpp#L42) set in libfm-qt.

**Comment by yan12125** [comment]
> But if lxqt-panel can still be compiled with menu-cache, the replacement won't be so easy.

lxqt-panel can be used with or without menu-cache via the CMake option `USE_MENU_CACHE`. The default has been OFF since 2018 as there is `no noticable penalty even on a 2004 single core pentium if not used` [1]. I think menu-cache codes in lxqt-panel can be dropped and lxqt-panel can just use https://github.com/stefonarch/lxqt-menu-data directly.

> `sudo rm /usr/share/desktop-directories/lxde-*`

This may not be feasible for now, as libfm-qt still uses lxmenu-data via menu-cache.

[1] https://github.com/lxqt/lxqt-panel/commit/d7a441cc7150b73adae1cf69114f5d8e3ac87a15

**Comment by tsujan** [comment]
> I think menu-cache codes in lxqt-panel can be dropped

I agree completely.

**Comment by stefonarch** [comment]
> > `sudo rm /usr/share/desktop-directories/lxde-*`
> 
> This may not be feasible for now, as libfm-qt still uses lxmenu-data via menu-cache.

I didn't find issues atm, https://github.com/lxqt/pcmanfm-qt/issues/952 is not reprodible also. But there may be hidden ones.

**Comment by yan12125** [comment]
> Looks like starting once menu-cached after the change fixed folder names and icons in menu://applications

After four steps in https://github.com/lxqt/lxqt/issues/1679#issuecomment-1450815984 and running menu-cached, I still have something like the screenshot in https://github.com/lxqt/lxqt-panel/pull/1869. Probably you got names and icons from previously cached menus in `~/.cache/menus/`? Try delete them and restart pcmanfm-qt.

> But here I see a [prefix to lxde](https://github.com/lxqt/libfm-qt/blob/d79919165fbda4b6a9cd1c49e3eceee69a127900/src/appmenuview.cpp#L42) set in libfm-qt.

Looks like that code is used in app chooser dialog only (File Properties -> Open With -> Customize). After force removing lxmenu-data and clear `~/.cache/menus/`, the app chooser shows nothing.

**Comment by tsujan** [comment]
Yes, the app chooser dialog uses men-cache.

As I've said [elsewhere](https://github.com/lxqt/libfm-qt/issues/417#issuecomment-1432470392), in my opinion, the removal of menu-cache shouldn't mean depending on another library — no even an LXQt library — because `libfm-qt` should remain modular and without extra dependencies.

**Comment by stefonarch** [comment]
> > But here I see a [prefix to lxde](https://github.com/lxqt/libfm-qt/blob/d79919165fbda4b6a9cd1c49e3eceee69a127900/src/appmenuview.cpp#L42) set in libfm-qt.
> 
> Looks like that code is used in app chooser dialog only (File Properties -> Open With -> Customize). After force removing lxmenu-data and clear `~/.cache/menus/`, the app chooser shows nothing.

I commented out those
```
//    qputenv("XDG_MENU_PREFIX", "lxde-");
    menu_cache = menu_cache_lookup("applications.menu");
    // if(!oldenv.isEmpty())
//    qputenv("XDG_MENU_PREFIX", oldenv); // restore the original value if needed

```
After compiling and restarting the desktop app-chooser dialog was populated again. So it looks that  `menu-cache` is working with `lxqt-menu-data` and without `lxmenu-data`.

No clue for "Applications" in pcmanfm-qt: it was only once without translations and icons and never again.

**Comment by tsujan** [comment]
> I commented out those

Very interesting! It didn't occur to me. Thanks for the info! I'll look into it.

**Comment by stefonarch** [comment]
> Yes, the app chooser dialog uses `menu-cache`.

What I don't really understand is why the panel doesn't use menu-cache anymore but in this dialog it's needed? Menu use in panel is much more frequent use case.

**Comment by tsujan** [comment]
Panel uses `libqtxdg`. That's not good for `libfm-qt` (https://github.com/lxqt/libfm-qt/issues/417#issuecomment-1432470392).

**Comment by yan12125** [comment]
Sounds like there is a feasible road to replace lxmenu-data. The first step might be transferring https://github.com/stefonarch/lxqt-menu-data to the https://github.com/lxqt organization?

**Comment by stefonarch** [comment]
Done: https://github.com/lxqt/lxqt-menu-data

**Comment by stefonarch** [comment]
Afaik the only new commit is the update of german translations. The question is if we are going to implement yaml/weblate for new/changing translations or just leave this to PRs. 
 If using `lxqt-settings-system.directory.in` the translations are already provided by  `yaml` files.

**Comment by yan12125** [comment]
It will be great if https://github.com/lxqt/lxqt-menu-data is also integrated with weblate. It's not a top priority; can be tackled with after the migration.

**Comment by yan12125** [comment]
I have another look at the `lxqt-settings-system.directory.in` situation. For now it's used only by `lxqt-config.menu`. Nevertheless, it may be good to move all `lxqt-settings-*.directory` to lxqt-menu-data, so that they can be used in other places (ex: https://github.com/lxqt/lxqt-panel/pull/1869/files#r1141262774).

**Comment by stefonarch** [comment]
I've been retesting all, it looks working. IMO [another menu file](https://github.com/lxqt/lxqt-menu-data/issues/2) is needed used  only with libfm-qt which provides a clean applications-menu without "Preferences", leave menu and lxqt-about.

**Comment by stefonarch** [comment]
Those are the  3 PRs  to close this issue - if I made no mistakes. I tested all of it:

https://github.com/lxqt/lxqt-config/pull/935
https://github.com/lxqt/lxqt-panel/pull/1869
https://github.com/lxqt/lxqt-menu-data/pull/4

Once merged another PR is needed for `libfm.qt`.

**Comment by yan12125** [comment]
All 3 PRs look good. Can we merge them now or we should wait for a PR for libfm-qt?

**Comment by stefonarch** [comment]
Thanks for reviewing.  
If I'm not mistaken as far as git users don't remove `lxmenu-data`  from the system there shouldn't be issues with application menus in pcmanfm-qt, so the PR is'nt urgent.

PKGBUILDs from AUR should be updated too after and one added for this.

**Comment by tsujan** [comment]
> Can we merge them now or we should wait for a PR for libfm-qt?

Since `libfm-qt` doesn't use it, it isn't an obstacle to merging.

After it's merged, I'll make a PR for `libfm-qt` to use it instead of `lxmenu-data`.

**Comment by yan12125** [comment]
Thanks! All 3 PRs are merged. A new AUR package lxqt-menu-data-git is uploaded and lxqt-config-git is updated. I'm not a maintainer of lxqt-panel-git, so I only leave a comment for now. If @pmattern (the AUR package maintainer) is too busy to update it recently, I can use my Trusted User priviledge to update it.

**Comment by tsujan** [comment]
Don't we want to make lxqt-panel and lxqt-config depend on `lxqt-menu-data` explicitly? We discussed it somewhere (I can't find it now) and you agreed.

**Comment by stefonarch** [comment]
It's here https://github.com/lxqt/lxqt-menu-data/pull/3#issuecomment-1523281385

I didn't include it because cmake syntax isn't my thing, but I'm not against it to have it as build dependency. BTW it should be included in the build scripts too, and I'll update the wiki now.

**Comment by yan12125** [comment]
Oops I forgot about that. Let me check if I can add it to CMakeLists.txt files.

**Comment by stefonarch** [comment]
I added the 4 categories to weblate https://translate.lxqt-project.org/projects/lxqt-configuration/ but what I don't understand is that the files from `lxqt-config` (other settings, system settings) are still present while the one from `lxqt-panel` has the warning symbol about missing sources as it should be. 

Atm they are doubled, of course I could delete them but I'd like to understand...

**Comment by stefonarch** [comment]
Only thing missing is adding the repo to the bot and testing translations.

**Comment by yan12125** [comment]
> Oops I forgot about that. Let me check if I can add it to CMakeLists.txt files.

My first time on playing with CMake dependencies. Hopefully I'm doing correct things :)

https://github.com/lxqt/lxqt-menu-data/pull/8
https://github.com/lxqt/lxqt-panel/pull/1892
https://github.com/lxqt/lxqt-config/pull/941

**Comment by stefonarch** [comment]
Oh  I missed it apparently, done now. I'm happy when this all is over :)

**Comment by tsujan** [comment]
@stefonarch, @yan12125, thanks for good work!
